compiler/rustc_hir_analysis/src/check/fn_ctxt/suggestions.rs

```diff
@@ -80,6 +80,38 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         let Some((def_id_or_name, output, inputs)) = self.extract_callable_info(expr, found)
             else { return false; };
         if can_satisfy(output) {
+            // suggest `unwrap_or_else` for `Option` and `Result`, when a closure is given
+            if let hir::ExprKind::Closure(..) = expr.kind &&
+                let parent = self.tcx.hir().get_parent_node(expr.hir_id) &&
+                let Some(Node::Expr(call_expr)) = self.tcx.hir().find(parent) &&
+                let hir::ExprKind::MethodCall(
+                      hir::PathSegment {ident: method_name, ..},
+                      receiver,
+                      _args,
+                      _span) = call_expr.kind &&
+                inputs.len() == 0 {
+              let self_ty = self.typeck_results.borrow().expr_ty(receiver);
+              let new_name = Ident {
+                  name: Symbol::intern(&format!("{}_else", method_name.as_str())),
+                  span: method_name.span,
+              };
+
+              if self.method_exists(
```

Comment:
I would use `lookup_probe` instead, so you can give it this probe scope: https://doc.rust-lang.org/nightly/nightly-rustc/rustc_hir_analysis/check/method/probe/enum.ProbeScope.html#variant.TraitsInScope so we don't suggest methods that are out of scope

Reply:
`method_exists` also use `TraitsInScope`, https://github.com/rust-lang/rust/blob/ab37a83ae2c829b8597bdcda8244964ebc60b495/compiler/rustc_hir_analysis/src/check/method/mod.rs#L104

I changed to use `lookup_probe` to get the pick result, and check the function arguments..